go.mod

```
@@ -1,15 +1,17 @@
 module github.com/scality/vaultclient-go
 
-go 1.18
+go 1.25
```

seems weird that we _require_ golang 1.25 : this field is used to indicate the _minimum_ supported version, not the "target" version. Basically we should not modify it unless we change the code in some incompatible way; and most of the time it would be automatically bump'ed when updating dependencies (e.g. with `go mod tidy` etc)

---

with recent version, you can additionally add a `toolchain` directive which defines the specific version which should (and is by default) be used to build the project


```suggestion
go 1.18         # <<--- Not sure we are still compatible with 1.18, but you get the gist

toolchain go1.25.0
```

---

Yeah, fixed it